kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -587,37 +669,97 @@ define([
+                delay: {
+                    show: Config.get('tooltip').showDelay,
+                    hide: Config.get('tooltip').hideDelay
+                }
+            })
+            .off('click');
+        },
+
+        enableImportButton: function() {
+            let stagingAreaViewer = this; 
+
+            this.$elem.find('button.kb-staging-table-import__button')
+            .removeClass('kb-staging-table-import__button__disabled')
+            .tooltip('disable')
+            .unbind('click')
+            .on('click keyPress', function() {
+                stagingAreaViewer.initImport();
+            });
+        },
+
+        initImport: function() {
+            let stagingAreaViewer = this; 
+
+            //get all of the selected checkbox file names and import type
+            $('input.kb-staging-table-body__checkbox-input:checked')
+            .each(function () {
+                const importType = $(this).attr('data-type');
+                const importFile = $(this).attr('data-file-name');
+                stagingAreaViewer.initImportApp(importType, importFile);
+            });
+          
+            stagingAreaViewer.updateView();
+        },
+
         /**
          * Initializes an import app using the given file info as input.
          * Expects 'type' to match a KBase object type string that maps onto an importer.
-         * Expects 'file' to be an object with the following attributes:
-         *   name = string, name of the file
+         * Expects 'file' to be a string that is the name of the file
          */
         initImportApp: function (type, file) {
-            var appInfo = this.uploaders.app_info[type];
+            const appInfo = this.uploaders.app_info[type];
+
             if (appInfo) {
-                var tag = APIUtil.getAppVersionTag(),
-                    fileParam = file ? file.name : '',
+                let tag = APIUtil.getAppVersionTag(),
+                    fileParam = file || '',
                     inputs = {};
+
                 if (this.subpath) {
-                    fileParam = this.subpath + '/' + file.name;
+                    fileParam = this.subpath + '/' + file;
                 }
+
                 if (appInfo.app_input_param_type && appInfo.app_input_param_type === 'list') {
                     fileParam = [fileParam];
                 }
+
                 if (appInfo.app_input_param) {
                     inputs[appInfo.app_input_param] = fileParam;
                 }
+
                 if (appInfo.app_output_param) {
-                    inputs[appInfo.app_output_param] = file.name.replace(/\s/g, '_') + appInfo.app_output_suffix;
+                    inputs[appInfo.app_output_param] = file.replace(/\s/g, '_') + appInfo.app_output_suffix;
                 }
+
                 if (appInfo.app_static_params) {
-                    for (var p in appInfo.app_static_params) {
+                    for (let p in appInfo.app_static_params) {
```

Comment:
Could update this to something like:
```
for (const p of Object.keys(appInfo.app_static_params)) {
    inputs[p] = appInfo.app_static_params[p];
}
```

or something similar. That for..in..if(hasOwnProperty) style is pretty outdated.